okhttp/src/main/java/io/grpc/transport/okhttp/OkHttpWritableBufferAllocator.java

```diff
@@ -38,8 +38,24 @@
  * The default allocator for {@link OkHttpWritableBuffer}s used by the OkHttp transport.
  */
 class OkHttpWritableBufferAllocator implements WritableBufferAllocator {
+
+  private final int maxDataLength;
+
+  /**
+   * Construct a new instance
+   * @param maxDataLength the maximum allowed length of a DATA frame.
```

Comment:
This largely means it should handle buffers > maxframesize. We still need an upper bound here however which can still be 4k for the moment.